internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/controller/VeniceParentHelixAdminTest.java

```diff
@@ -538,24 +537,19 @@ public void testSupersetSchemaWithCustomSupersetSchemaGenerator() throws IOExcep
     }
   }
 
-  @DataProvider(name = "CONTROLLER_SSL_SUPERSET_SCHEMA_GENERATOR")
-  public static Object[][] controllerSSLAndSupersetSchemaGenerator() {
-    return new Object[][] { new Object[] { true, Optional.of(new SupersetSchemaGeneratorWithCustomProp("test_prop")) },
-        new Object[] { false, Optional.empty() } };
-  }
-
-  @Test(dataProvider = "CONTROLLER_SSL_SUPERSET_SCHEMA_GENERATOR", timeOut = DEFAULT_TEST_TIMEOUT_MS * 10)
-  public void testStoreMetaDataUpdateFromParentToChildController(
-      boolean isControllerSslEnabled,
-      Optional<SupersetSchemaGenerator> supersetSchemaGenerator) throws IOException {
+  @Test(dataProvider = "True-and-False", dataProviderClass = DataProviderUtils.class, timeOut = DEFAULT_TEST_TIMEOUT_MS
+      * 10)
+  public void testStoreMetaDataUpdateFromParentToChildController(boolean isControllerSslEnabled) throws IOException {
     String clusterName = Utils.getUniqueString("testStoreMetadataUpdate");
     Properties properties = new Properties();
     // This cluster setup don't have server, we cannot perform push here.
     properties.setProperty(CONTROLLER_AUTO_MATERIALIZE_META_SYSTEM_STORE, String.valueOf(false));
     properties.setProperty(CONTROLLER_AUTO_MATERIALIZE_DAVINCI_PUSH_STATUS_SYSTEM_STORE, String.valueOf(false));
-    if (supersetSchemaGenerator.isPresent()) {
+    if (isControllerSslEnabled) {
       properties.setProperty(CONTROLLER_PARENT_EXTERNAL_SUPERSET_SCHEMA_GENERATION_ENABLED, String.valueOf(true));
-      properties.put(VeniceControllerWrapper.SUPERSET_SCHEMA_GENERATOR, supersetSchemaGenerator.get());
+      properties.put(
```

Comment:
I am not very sure why ssl enabled also means supserschemagenerator enabled. Although previously supersetSchemaGenerator will only be enabled when ssl enabled = true.

Reply:
Yeah current settings only test sslEnabled + generator, and sslDisabled + no generator. But I understand that using `isControllerSslEnabled` to control superset schema generator configs makes the code looks confusing.

The idea is: do not use SupersetSchemaGenerator object as a test parameter. Because when test retries, the new object's string representation will be different (e.g. SupersetSchemaGeneratorWithCustomProp@15ad1181 vs. SupersetSchemaGeneratorWithCustomProp@2839b184). Even though retry passes, the test will be regarded as a failure without a retry success. Entire CI will be marked as failure.

Another way to fix the problem is to change SupersetSchemaGenerator object to a boolean enableSuperSetSchemaGeneration and only enable superset schema generator when the boolean is true.

Reply:
Rebase, resolve conflicts, and use a boolean parameter to control superset schema generator configs in this test. Please take another look when you have time.